base/set.jl

```diff
@@ -201,6 +201,109 @@ function unique(f::Callable, C)
     out
 end
 
+# If A is not sorted, then we will need to keep track of all of the elements that we have
+# seen so far.
+function _unique!(A)
+    seen = Set{eltype(A)}()
+    idxs = eachindex(A)
+    i = state = start(idxs)
+    for x in A
+        if x ∉ seen
+            push!(seen, x)
+            i, state = next(idxs, state)
+            A[i] = x
+        end
+    end
+    resize!(A, i - first(idxs) + 1)
+end
+
+# If A is sorted, then we only need to keep track of one element at a time. We replace the
+# elements of A with the unique elements that we see in the order that we see them. Once
+# we have iterated through A, we resize A based on the number of unique elements that we
+# see.
+function _sortedunique!(A)
```

Comment:
Even if it is obvious from the context, I think it would help readers to make some assumptions explicit, e.g. `A::AbstractArray`, and that a precondition is that `!isempty(A)`.